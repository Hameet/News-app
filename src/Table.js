import React, { Component } from 'react'

function isSearched (searchTerm) {
  return function (item) {
    return item.author.toLowerCase().includes(searchTerm.toLowerCase())
  }
}

const Button = ({ onClick, className = '', children }) => {
  return (
    <button onClick={onClick} className={className} type='button'>
      {children}
    </button>
  )
}

const Table = ({ list, pattern, onDismiss }) => {
  return (
    <div>
      {list.filter(isSearched(pattern)).map(item => {
        return (
          <div key={item.objectID}>

            <span>
              <a href={item.url}>{item.title}</a>
            </span>

            <span>{item.author}</span>
            <span>{item.num_comments}</span>
            <span>{item.points}</span>
            <span>
              <Button onClick={() => onDismiss(item.objectID)}>
                Dismiss
              </Button>
            </span>
          </div>
        )
      })}
    </div>
  )
}
export default Table
