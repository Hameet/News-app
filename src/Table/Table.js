import React, { Component } from 'react'

import Button from '../Button/Button'

const largeColumn = {
  width: '50%'
}
const midColumn = {
  width: '20%'
}
const smallColumn = {
  width: '10%'
}

const Table = ({ list, onDismiss }) => {
  return (
    <div className='table'>
      {list.map(item => {
        return (
          <div key={item.objectID} className='table-row'>
            <span style={largeColumn}>
              <h3><a href={item.url}>{item.title}</a></h3>
            </span>
            <span style={midColumn}>
              <h3>{item.author}</h3>
            </span>
            <span style={smallColumn}>
              {item.num_comments}
            </span>
            <span style={smallColumn}>
              {item.points}
            </span>
            <span style={smallColumn}>
              <Button onClick={() => onDismiss(item.objectID)}>
                Dismiss
              </Button>
            </span>
          </div>
        )
      })}
    </div>
  )
}
export default Table
