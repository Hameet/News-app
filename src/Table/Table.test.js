import React from 'react'
import ReactDOM from 'react-dom'
import Enzyme, { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Table from './Table'
import Button from '../Button/Button'

describe('Table', () => {
  const props = {
    list: [
      { title: '1', author: '1', num_comments: 1, points: 2, objectID: 'y' },
      { title: '2', author: '2', num_comments: 1, points: 2, objectID: 'z' }
    ]
  }

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Table {...props}>Table</Table>, div)
  })
  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Table {...props}><Button>Dismiss</Button></Table>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows two items in list', () => {
    const element = shallow(<Table {...props} />)
    expect(element.find('.table-row').length).toBe(2)
  })

  it('shows button is present', () => {
    const element = shallow(<Table {...props} />)
    expect(element.find(<Button />).length).toBe(0)
  })
})
