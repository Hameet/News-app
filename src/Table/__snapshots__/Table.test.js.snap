// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table has a valid snapshot 1`] = `
<div
  className="table"
>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <h3>
        <a>
          1
        </a>
      </h3>
    </span>
    <span
      style={
        Object {
          "width": "20%",
        }
      }
    >
      <h3>
        1
      </h3>
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      1
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      2
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        Dismiss
      </button>
    </span>
  </div>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <h3>
        <a>
          2
        </a>
      </h3>
    </span>
    <span
      style={
        Object {
          "width": "20%",
        }
      }
    >
      <h3>
        2
      </h3>
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      1
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      2
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        Dismiss
      </button>
    </span>
  </div>
</div>
`;
